### PR TITLE
feature/Store

### PR DESCRIPTION
**Pull request by OGNova**
This PR features modifications to the store command, and general fixes.

### CODE REVIEW

**Review by 7bitlyrus — CHANGES_REQUESTED** [review]
Checking for both the name of a role and the ID of a role is pointless and may be an issue. You should change these checks to just a check of the ID, for example:
`check: message => message.member.roles.has(roleid)`

File: commands/exec.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ class Exec extends Command {
       category: 'System',
       extended: 'This will spawn a child process and execute the given command.',
       botPerms: ['SEND_MESSAGES'],
-      permLevel: 'Bot Owner'
+      permLevel: 'Police'
```
Comment (7bitlyrus) [inline commands/exec.js]: Not completely sure if it's a wise idea to allow all police to use exec, but I'm not sure.

Comment (OGNova) [reply commands/exec.js]: I can restructure the perm levels, but then I'd need IDs of users that you'd like to be allowed to use exec.

File: util/setup_base.txt [hunk 2]
```diff
@@ -40,60 +42,49 @@ const config = {
...
 
     { level: 3,
-      name: "Administrator", 
+      name: 'Council', 
       check: (message) => {
         try {
-          const adminRole = message.guild.roles.find(r => r.name.toLowerCase() === message.settings.adminRole.toLowerCase());
-          return (adminRole && message.member.roles.has(adminRole.id));
+          const councilRole = message.guild.roles.find(r => r.name.toLowerCase() === message.settings.councilRole.toLowerCase());
+          return (councilRole && message.member.roles.has(councilRole.id));
```
Comment (slice) [inline util/setup_base.txt]: This entire block can be replaced with:

```js
message.member.roles
  .find(role => role.name.toLowerCase(), message.settings.councilRole.toLowerCase())
```

File: commands/exec.js (hunk 1 above)
Comment (7bitlyrus) [reply commands/exec.js]: It would be a wise idea to leave it open to only the host of the bot and configurable.

File: util/setup_base.txt [hunk 3]
```diff
@@ -40,60 +42,49 @@ const config = {
...
       // The following lines check the guild the message came from for the roles.
       // Then it checks if the member that authored the message has the role.
       // If they do return true, which will allow them to execute the command in question.
       // If they don't then return false, which will prevent them from executing the command.
       check: (message) => {
         try {
-          const modRole = message.guild.roles.find(r => r.name.toLowerCase() === message.settings.modRole.toLowerCase());
-          if (modRole && message.member.roles.has(modRole.id)) return true;
+          const traineeRole = message.guild.roles.find(r => r.name.toLowerCase() === message.settings.traineeRole.toLowerCase());
+          if (traineeRole && message.member.roles.has(traineeRole.id)) return true;
```
Comment (slice) [inline util/setup_base.txt]: This entire block can be replaced with:

```js
message.member.roles
  .find(role => role.name.toLowerCase(), message.settings.traineeRole.toLowerCase())
```

File: util/setup_base.txt [hunk 4]
```diff
@@ -40,60 +42,49 @@ const config = {
...
-
     // Bot Admin has some limited access like rebooting the bot or reloading commands.
     { level: 9,
-      name: "Bot Admin",
-      check: (message) => config.admins.includes(message.author.id)
+      name: 'Police', 
+      check: (message) => {
+        try {
+          const policeRole = message.guild.roles.find(r => r.name.toLowerCase() === message.settings.policeRole.toLowerCase());
+          return (policeRole && message.member.roles.has(policeRole.id));
```
Comment (slice) [inline util/setup_base.txt]: This entire block can be replaced with:

```js
message.member.roles
  .find(role => role.name.toLowerCase(), message.settings.policeRole.toLowerCase())
```

**Review by slice — COMMENTED** [review]
This code can also be further condensed with a Higher-Order Function (function factory).

File: util/setup_base.txt (hunk 3 above)
Comment (OGNova) [reply util/setup_base.txt]: Changing the way those checks are being done currently.

Comment (slice) [reply util/setup_base.txt]: :+1:

**Review by 7bitlyrus — COMMENTED** [review]
👍

**Review by 7bitlyrus — DISMISSED** [review]
Looks good, thanks!

File: util/setup_base.js [hunk 5]
```diff
@@ -40,60 +36,49 @@ const config = {
...
-
     // Bot Admin has some limited access like rebooting the bot or reloading commands.
     { level: 9,
-      name: "Bot Admin",
-      check: (message) => config.admins.includes(message.author.id)
+      name: 'Police', 
+      check: (message) => {
+        try {
+          const config = require('./config.js');
+          if (message.member.roles.has(config.policeRole)) return true;
```
Comment (slice) [inline util/setup_base.js]: `return message.member.roles.has(config.policeRole);`

Comment (slice) [reply util/setup_base.js]: Also, this try/catch block could be redundant.

**Review by slice — CHANGES_REQUESTED** [review]

**Review by ThaTiemsz — DISMISSED** [review]
`setup.js` should be updated to reflect the file extension change of `setup_base.js`

File: util/setup_base.js (hunk 5 above)
Comment (OGNova) [reply util/setup_base.js]: Should be good to go.

**Review by slice — DISMISSED** [review]

**Review by ThaTiemsz — APPROVED** [review]

**Pull request merged**